ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -766,6 +767,28 @@ static int string2attr_list(char *str, struct attr_value_list **__av_list,
 	return rc;
 }
 
+int validate_ldmsd_req(ldmsd_req_ctxt_t reqc, ldmsd_req_hdr_t rh)
+{
+	ldmsd_req_attr_t attr;
+	unsigned int off;
+	/* Validate length of request header and request. */
+	if (rh->rec_len <= sizeof(*rh))
+		return 0;
+	attr = (ldmsd_req_attr_t)(rh+1);
+	while (attr->discrim)
```

Comment:
We have to verify that the request buffer is large enough to hold this attr struct _before_ we access anything in the attr struct (i.e before accessing attr->discrim).

Reply:
The code still need this. Must check that "off+sizeof(*attr) >= rh->rec_len" _before_ accessing things in the attr.